src/code-checks/code-checks.service.ts

```diff
@@ -0,0 +1,324 @@
+import { SorbentTrapImportDTO } from "../dto/sorbent-trap.dto";
+import { Nsps4tSummaryImportDTO } from "../dto/nsps4t-summary.dto";
+import { FuelCode } from "../entities/fuel-code.entity";
+import { EaseyException } from "@us-epa-camd/easey-common/exceptions/easey.exception";
+import { TestTypeCode } from "../entities/test-type-code.entity";
+import { TestResultCode } from "../entities/test-result-code.entity";
+import { SpanScaleCode } from "../entities/span-scale-code.entity";
+import { GasLevelCode } from "../entities/gas-level-code.entity";
+import { ProtocolGasVendor } from "../entities/protocol-gas-vendor.entity";
+import { InjectionProtocolCode } from "../entities/injection-protocol-codes.entity";
+import { UnitsOfMeasureCode } from "../entities/units-of-measure.entity";
+import { ModcCode } from "../entities/modc-code.entity";
+import { OperatingConditionCode } from "../entities/operating-condition-code.entity";
+import { SodMassCode } from "../entities/sod-mass-code.entity";
+import { SodVolumetricCode } from "../entities/sod-volumetri-code.entity";
+import { SampleTypeCode } from "../entities/sample-type-code.entity";
+import { BeginEndHourFlag } from "../entities/begin-end-hour-flag.entity";
+import { FuelFlowPeriodCode } from "../entities/fuel-flow-period-code.entity";
+import { ApsCode } from "../entities/aps-code.entity";
+import { TrainQaStatusCode } from "../entities/train-qa-status-code.entity";
+import { Nsps4tElectricalLoadCode } from "../entities/nsps4t-electrical-load-code.entity";
+import { Nsps4tEmissionStandardCode } from "../entities/nsps4t-emission-standard-code.entity";
+
+const getErrorMessage = (property, value) => `You reported an invalid ${property} of ${value}.`;
+@Injectable()
+export class CodeChecksService {
+
+    constructor(private readonly logger: Logger) {
+        this.logger.setContext('CodeChecksService');
+    }
+
+    async runChecks(payload: EmissionsImportDTO): Promise<string[]> {
+        this.logger.log('Running Code Checks');
+        // NOTE: codesMap is not a class variable because providers are set to singleton scope so declaring codesMap as a class variable  will cause it to hold values for every request
+        const codesMap = new Map<string, Set<string>>();
+
+        const errorList = [];
+
+        // Build up codesMap
+        this.collectDailyEmissionsCodes(payload.dailyEmissionData, codesMap);
+        this.collectWeeklyTestSummaryCodes(payload.weeklyTestSummaryData, codesMap);
+        this.collectSummaryValueCodes(payload.summaryValueData, codesMap);
+        this.collectDailyTestSummary(payload.dailyTestSummaryData, codesMap);
+        this.collectHourlyOperatingData(payload.hourlyOperatingData, codesMap);
+        this.collectLongTermFuelFlowData(payload.longTermFuelFlowData, codesMap);
+        this.collectSorbentTrapData(payload.sorbentTrapData, codesMap);
+        this.collectNsps4tSummary(payload.nsps4tSummaryData, codesMap);
+
+        // Asynchronously make the DB queries
```

Comment:
this is just a little cleaner way of handling promises but consider pushing each promise into a promise array and setting a const to track its index to be used later.

let index = 0;
const promises = [];

do the following for each one...
const PARAMETER_CODES= index++;
promises.push(getInvalidCodes<ParameterCode>(codesMap.get("parameterCode"), ParameterCode.getRepository()));